test/unittest/statistics/rtps/MonitorServiceTests.cpp

```diff
@@ -0,0 +1,274 @@
+                dds::SampleLostStatus& status));
+
+};
+
+struct MockConnectionsQueryable : public IConnectionsQueryable
+{
+    MOCK_METHOD1(get_entity_connections, ConnectionList(
+                const fastrtps::rtps::GUID_t& guid));
+};
+
+struct MockProxyQueryable : public IProxyQueryable
+{
+
+    MOCK_METHOD1(get_all_local_proxies, bool(
+                std::vector<fastrtps::rtps::GUID_t>& guids));
+
+    MOCK_METHOD2(get_serialized_proxy, bool(
+                const fastrtps::rtps::GUID_t& guid,
+                fastrtps::rtps::CDRMessage_t* msg));
+};
+
+
+class MonitorServiceTests : public ::testing::Test
+{
+
+public:
+
+    MonitorServiceTests()
+        : monitor_srv_(
+            fastrtps::rtps::GUID_t(),
+            &mock_proxy_q_,
+            &mock_conns_q_,
+            mock_status_q_)
+        , listener_(&monitor_srv_)
+        , n_local_entities(5)
+    {
+
+    }
+
+    void SetUp() override
+    {
+        mock_guids.reserve(n_local_entities);
+
+        ON_CALL(mock_proxy_q_, get_all_local_proxies(::testing::_)).WillByDefault(testing::Invoke(
+                    [this](std::vector<fastrtps::rtps::GUID_t>& guids)
+                    {
+                        guids.reserve(n_local_entities);
+                        mock_guids.reserve(n_local_entities);
+
+                        for (size_t i = 1; i <= static_cast<size_t>(n_local_entities); i++)
+                        {
+                            fastrtps::rtps::GUID_t guid;
+                            guid.entityId.value[3] = i;
+                            guids.push_back(guid);
+                            mock_guids.push_back(guid);
+                        }
+                        return true;
+                    }));
+    }
+
+    void TearDown() override
+    {
+
+    }
+
+protected:
+
+    MockConnectionsQueryable mock_conns_q_;
+    MockStatusQueryable mock_status_q_;
+    MockProxyQueryable mock_proxy_q_;
+
+    MonitorService monitor_srv_;
+    MonitorServiceListener listener_;
+    int n_local_entities;
+    std::vector<fastrtps::rtps::GUID_t> mock_guids;
+};
+
+TEST_F(MonitorServiceTests, enabling_monitor_service_routine)
+{
+    //! At the startup, the service should collect the already existing
+    //! local entities
+    EXPECT_CALL(mock_proxy_q_, get_all_local_proxies(::testing::_)).Times(1);
+    EXPECT_CALL(mock_proxy_q_, get_serialized_proxy(::testing::_, ::testing::_)).Times(n_local_entities);
+    EXPECT_CALL(mock_conns_q_, get_entity_connections(::testing::_)).Times(n_local_entities);
+
+    //! Enable the service
+    ASSERT_FALSE(monitor_srv_.disable_monitor_service());
+    ASSERT_TRUE(monitor_srv_.enable_monitor_service());
+    ASSERT_TRUE(monitor_srv_.is_enabled());
+
+    //! Verify expectations
+    std::this_thread::sleep_for(std::chrono::seconds(3));
+    ::testing::Mock::VerifyAndClearExpectations(&mock_conns_q_);
+    ::testing::Mock::VerifyAndClearExpectations(&mock_proxy_q_);
```

Comment:
Should we ASSERT_TRUE these just in case?

Reply:
I am checking other tests and it is not added. But it think we should

Reply:
Otherwise we would just be clearing the expectations, not verifying them.